### PR TITLE
Improved compatibility suite to handle unordered results and prettified path expressions

### PR DESCRIPTION
**Pull request by craigtaverner**

### CODE REVIEW

**Comment by davidegrohmann** [comment]
Based on top of https://github.com/neo4j/neo4j/pull/4594

File: community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/MatchAcceptanceTest.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ import scala.collection.JavaConverters._
 
 class MatchAcceptanceTest extends ExecutionEngineFunSuite with QueryStatisticsTestSupport with NewPlannerTestSupport {
 
-  test("path query should return results in written order") {
+  ignore("path query should return results in written order") {
```
Comment (Mats-SX) [inline community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/MatchAcceptanceTest.scala]: I think these `ignore` changes are good to have for knowing which tests have been ported and which haven't, but we need to recall to revert these changes back before merging this.
